design/design_stopping_sims.Rmd

```diff
@@ -0,0 +1,229 @@
+
+```{r, eval = F}
+stopLength <-length(object@stopping@stopList)
+                stop_sum <- matrix(nrow=1,ncol=stopLength)
+                for (i in 1: stopLength){
+                  stop_sum[,i]<- stopTrial(stopping=object@stopping@stopList[[i]],
+                                           dose=thisDose,
+                                           samples=thisSamples,
+                                           model=object@model,
+                                           data=thisData)
+                  
+                }
+
+```
+As a result we would get a matrix for each iteration that looks like this: 
+
+
+```{r}
+output_it1 <- matrix(nrow= 1, ncol = 2)
+output_it2 <- matrix(nrow= 1, ncol = 2)
+output_it3 <- matrix(nrow= 1, ncol = 2)
+output_it4 <- matrix(nrow= 1, ncol = 2)
+output_it1[1, ] <- c(TRUE, FALSE)
+output_it2[1, ] <- c(FALSE, TRUE)
+output_it3[1, ] <- c(TRUE, FALSE)
+output_it4[1, ] <- c(TRUE, TRUE)
+
+output <- list(output_it1, output_it2, output_it3, output_it4)
+output
+```
+
+Based on this we calculate the measure we're interested in: stopping reasons for all stopping rules and the first hit per iteration (that would happen in the summary method):
+```{r}
+stopReasonsUnique <- sapply(output,colSums)
+stopReasonsUnique
+stopReasonsUniquePercent<- (rowSums(stopReasonsUnique)/length(stopReasonsUnique[1,]))*100
+stopReasonsUniquePercent
+              
+
+## %trials each stopping rule fired first based on the order of the stopping rule defined in the design
+stopReasonsFirstHit<-apply(stopReasonsUnique,2,function(x){min(which(x == "1"))})
+stopReasonsFirstHitFreq<-table(stopReasonsFirstHit)
+
+stopReasonsFirstHit              
+stopReasonsFirstHitFreq              
+              
+if (dim(stopReasonsFirstHitFreq)<=length(output[[1]])){
+          frame<-as.data.frame(stopReasonsFirstHitFreq)
+          framefull<-data.frame(stopReasonsFirstHit=1:length(output[[1]]))
+          frame1 <- merge(x=framefull ,y= frame, by="stopReasonsFirstHit", all.x = TRUE)
+          frame1$Freq[is.na(frame1$Freq)] = 0
+          stopReasonsFirstHitFreq <- as.numeric(frame1$Freq)
+              }
+              
+## percentage of different values of the freq
+stopReasonsFirstHitFreqPercent <- (stopReasonsFirstHitFreq/sum(stopReasonsFirstHitFreq))*100
+stopReasonsFirstHitFreqPercent              
+```
+
+If we put everything together, the output can then for example look like this: 
+
+>>Stops (individual) for reason 1 & 2 StoppingMinCohorts and StoppingTargetProb : 75 %  
+>>Stops (individual) for reason 3 StoppingMinPatients : 50 %  
+
+>>Stops (first hit) for reason 1 & 2 SStoppingMinCohorts and StoppingTargetProb : 75 %  
+>>Stops (first hit) for reason 3 StoppingMinPatients : 25 %  
+
+
+
+## Class and Method functions affected 
+
+* method "simulate"
```

Comment:
you can also here include prototype code by using another (similar) generic function name, e.g.:

```r
setMethod("new_simulate", ...)
```